### PR TITLE
Implemented FormUrlEncodePropertyAttribute and it's usage

### LINKED ISSUE #90: Attribute to set name to form url encoded parameters
Current options is to override the whole content serializer. It would be nice if there was a build in attribute to use for this, like 
```
        public class TokenRequest
        {
            [FormParameter("username")]
            public string Username { get; set; }

            [FormParameter("password ")]
            public string Password { get; set; }

            [FormParameter("password")]
            public string GrantType { get; set; }

            [FormParameter("client_id ")]
            public string ClientId { get; set; }
        }
```

### PR DESCRIPTION
**Pull request by carl-berg**
Fixes #90
Enables custom form url encoded parameters like so
```
        public class CustomizedPayload
        {
            public int Id { get; set; }
            public string Name { get; set; }

            [FormUrlEncodedProperty("refresh_token")]
            public string RefreshToken => "my-token";
        }
```
in a similar way to how it's done with json using the `[JsonProperty("refresh_token")]`

### CODE REVIEW

**Comment by mattiasnordqvist** [comment]
Nice that it is named like newtonsofts counterpart. I think it starts to get a little hard to remember all the ways you can rename things now, with AliasAttribute for QueryParameters, JsonProperty and FormUrlEncodedProperty. And switching body content from formurlencoded to json would make currently applied attributes have no effect. Would it be possible to create a catch-all-attribute, lika `Alias`, to work for all situations? 

What do you think? Is this a problem or not really @carl-berg ? 👍 Otherwise code is fine and I'll prepare a new release for this along with some other minor tweaks.

**Comment by carl-berg** [comment]
A catch all would be nice i think, but would you want it to wok both for request and response or would you want to separate the two?

**Comment by carl-berg** [comment]
As for Json deserialization, a `ContractResolver` that does this might be a way to intercept and rename a property. https://www.newtonsoft.com/json/help/html/ContractResolver.htm

**Comment by mattiasnordqvist** [comment]
Let's leave it like this for now. To much work... :)

File: WebAnchor/RequestFactory/Serialization/FormUrlEncodedSerializer.cs [hunk 1]
```diff
@@ -14,8 +14,20 @@ public HttpContent Serialize(object value, Parameter content)
...
+        protected IEnumerable<KeyValuePair<string, string>> GetKeyValues(object content)
+        {
+            foreach (var property in content.GetType().GetProperties())
+            {
+                var name = property.GetCustomAttribute<FormUrlEncodedPropertyAttribute>(true)?.ParameterName ?? property.Name;
+                var value = property.GetGetMethod().Invoke(content, null)?.ToString() ?? string.Empty;
+                yield return new KeyValuePair<string, string>(name, value);
+            }
+
+            yield break;
```
Comment (mattiasnordqvist) [inline WebAnchor/RequestFactory/Serialization/FormUrlEncodedSerializer.cs]: `yield break` is not really needed here. Loop has already ended.

Comment (mattiasnordqvist) [reply WebAnchor/RequestFactory/Serialization/FormUrlEncodedSerializer.cs]: or is it @carl-berg ?

Comment (carl-berg) [reply WebAnchor/RequestFactory/Serialization/FormUrlEncodedSerializer.cs]: No, you're right it can be removed.

**Pull request merged**